RELEASE_NOTES.md

```diff
@@ -1,5 +1,16 @@
 # Unreleased Version
 
+# ChefDK 3.5 Release Notes
+
+## Chef 14.7.17
+
+ChefDK now ships with Chef 14.7.17\. See <https://docs.chef.io/release_notes.html> for more information on what's new.
```

Comment:
"\" probably doesn't below in this first sentence.

Reply:
Also, this forward slash is present in the previous release notes (see ln 18) so I may just not be aware of certain formatting notations...

Reply:
For some reason, we've got that period after the Chef version backslashed going back to the notes for ChefDK 3.0. Do we cargo cult much?

Reply:
I cannot think of any reason we need that in our markdown formatting.